src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -14,27 +16,46 @@
 import reactor.core.publisher.Mono;
 
 import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
 
 @Component
 public class PaymentGatewayClient {
     @Autowired
     @Qualifier("paymentTransactionGatewayPostepayWebClient")
     PostePayInternalApi paymentTransactionGatewayPostepayWebClient;
 
+    @Autowired
+    private ObjectMapper objectMapper;
+
 
     public Mono<PostePayAuthResponseEntityDto> requestAuthorization(AuthorizationRequestData authorizationData) {
         PostePayAuthRequestDto postePayAuthRequest = new PostePayAuthRequestDto()
-                .grandTotal(BigDecimal.valueOf(authorizationData.transaction().getAmount().value() + authorizationData.fee()))
+                .grandTotal(BigDecimal.valueOf(((long) authorizationData.transaction().getAmount().value()) + authorizationData.fee()))
                 .description(authorizationData.transaction().getDescription().value())
                 .paymentChannel(authorizationData.pspChannelCode())
                 .idTransaction(authorizationData.transaction().getTransactionId().value().toString());
 
-        return paymentTransactionGatewayPostepayWebClient.authRequest( postePayAuthRequest, false, authorizationData.transaction().getTransactionId().value().toString())
+        String encodedMdcFields = encodeMdcFields(authorizationData);
+
+        return paymentTransactionGatewayPostepayWebClient.authRequest(postePayAuthRequest, false, encodedMdcFields)
                 .onErrorMap(WebClientResponseException.class, exception -> switch (exception.getStatusCode()) {
                     case UNAUTHORIZED -> new AlreadyProcessedException(authorizationData.transaction().getRptId());
                     case GATEWAY_TIMEOUT -> new GatewayTimeoutException();
                     case INTERNAL_SERVER_ERROR -> new BadGatewayException("");
                     default -> exception;
                 });
     }
+
+    private String encodeMdcFields(AuthorizationRequestData authorizationData) {
+        String mdcData;
+        try {
+            mdcData = objectMapper.writeValueAsString(Map.of("ecommerceTransactionId", authorizationData.transaction().getTransactionId().value()));
```

Comment:
in the `pagopa-ecommerce-transaction-service` MDC there is `transactionId`, so we could leave `transactionId` in  MDC fileds send to `payment-transaction-gateway`. in this way on `AI` we could query `traces.customDimension.transactionId` and retrieve all the logs of that transaction, both of the `transaction-service` and  `payment-trassaction-gateway`.

Reply:
Fixed in [127241f](https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/73/commits/127241fea99a58f69f1268511cdf67c55c31446a)